server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -906,6 +922,43 @@ static Settings aggregateIndexSettings(
         return indexSettings;
     }
 
+    /**
+     * Updates index settings to enable remote store by default based on cluster level settings
+     * @param settingsBuilder index settings builder to be updated with relevant settings
+     * @param requestSettings settings passed in during index create request
+     * @param clusterSettings cluster level settings
+     */
+    private static void updateRemoteStoreSettings(Settings.Builder settingsBuilder, Settings requestSettings, Settings clusterSettings) {
+        if (FeatureFlags.isEnabled(FeatureFlags.REMOTE_STORE)
+            && FeatureFlags.isEnabled(FeatureFlags.REPLICATION_TYPE)
+            && CLUSTER_REMOTE_STORE_INDEX_FORCE_SETTING.get(clusterSettings)) {
+            // Verify if we can create a remote store based index based on user provided settings
+            if ((INDEX_REPLICATION_TYPE_SETTING.exists(requestSettings)
+                && !INDEX_REPLICATION_TYPE_SETTING.get(requestSettings).equals(ReplicationType.SEGMENT))
+                || (INDEX_REMOTE_STORE_ENABLED_SETTING.exists(requestSettings)
+                    && !INDEX_REMOTE_STORE_ENABLED_SETTING.get(requestSettings))) {
+                return;
+            }
+            String segmentStoreRepo = requestSettings.get(SETTING_REMOTE_STORE_REPOSITORY);
+            if (segmentStoreRepo == null) {
+                segmentStoreRepo = CLUSTER_REMOTE_STORE_DEFAULT_SEGMENT_REPO_SETTING.get(clusterSettings);
+            }
+            settingsBuilder.put(SETTING_REPLICATION_TYPE, ReplicationType.SEGMENT)
+                .put(SETTING_REMOTE_STORE_ENABLED, true)
+                .put(SETTING_REMOTE_STORE_REPOSITORY, segmentStoreRepo);
+
+            if (!Objects.equals(requestSettings.get(INDEX_REMOTE_TRANSLOG_STORE_ENABLED_SETTING.getKey()), "false")) {
```

Comment:
we also need to check if `translog_repo` is also supplied . The reason being either we pick both `enabled` and `repo` from index settings . If not provided , fall back to cluster level settings.

Reply:
do we not allow user to just send `INDEX_REMOTE_TRANSLOG_STORE_ENABLED_SETTING` as true and pick translog repo from cluster settings? I think we should allow this, let me know what you think